spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindHidingSubClass.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * SpotBugs - Find bugs in Java programs
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+package edu.umd.cs.findbugs.detect;
+
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.Detector;
+import edu.umd.cs.findbugs.SourceLineAnnotation;
+import edu.umd.cs.findbugs.ba.ClassContext;
+import org.apache.bcel.generic.BasicType;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.ba.AnalysisContext;
+import org.apache.bcel.classfile.JavaClass;
+import org.apache.bcel.classfile.Method;
+import org.apache.bcel.Const;
+import org.apache.bcel.generic.Type;
+
+import java.util.Arrays;
+
+/**
+ * This detector finds all the methods of a subclass which are hiding the static methods of the superclass and
+ * tries to invoke that method using the instance variable.
+ * Please see @see <a href="https://wiki.sei.cmu.edu/confluence/display/java/MET07-J.+Never+declare+a+class+method+that+hides+a+method+declared+in+a+superclass+or+superinterface">SEI CERT MET07-J</a>
+ * @author Nazir, Muhammad Zafar Iqbal
+ */
+public class FindHidingSubClass implements Detector {
```

Comment:
I think it will be more descriptive if you name the class `FindHiddenMethod` instead.

Reply:
Thank you for pointing out the miss-naming. I have renamed the checker now. I have also changed it at all the relevant places.